client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -112,16 +160,19 @@ private[sharing] case class ListAllTablesResponse(
 
 /** A REST client to fetch Delta metadata from remote server. */
 class DeltaSharingRestClient(
-    profileProvider: DeltaSharingProfileProvider,
-    timeoutInSeconds: Int = 120,
-    numRetries: Int = 10,
-    maxRetryDuration: Long = Long.MaxValue,
-    sslTrustAll: Boolean = false,
-    forStreaming: Boolean = false,
-    responseFormat: String = DeltaSharingRestClient.RESPONSE_FORMAT_PARQUET,
-    readerFeatures: String = "",
-    queryTablePaginationEnabled: Boolean = false,
-    maxFilesPerReq: Int = 100000
+      profileProvider: DeltaSharingProfileProvider,
+      timeoutInSeconds: Int = 120,
+      numRetries: Int = 10,
+      maxRetryDuration: Long = Long.MaxValue,
+      sslTrustAll: Boolean = false,
+      forStreaming: Boolean = false,
+      responseFormat: String = DeltaSharingRestClient.RESPONSE_FORMAT_PARQUET,
+      readerFeatures: String = "",
+      queryTablePaginationEnabled: Boolean = false,
+      maxFilesPerReq: Int = 100000,
+      enableAsyncQuery: Boolean = false,
```

Comment:
Is it possible to have cleaner separate on the code path to handle sync v.s. async queries? 

To make it easier to read and make sure the sync queries are not affected.